frontend/src/components/common/form/StepForm/context.tsx

```diff
@@ -23,7 +27,77 @@ export const StepperFormProvider: React.FC<IStepperFormProviderProps> = ({
   steps,
   tabs,
 }) => {
-  const { values, setFieldValue } = useFormikContext<ISteppedFormValues>();
+  const { values, setFieldValue, setErrors, setTouched } = useFormikContext<ISteppedFormValues>();
+
+  const activeTab = values.activeTab;
+
+  /**
+   * Get the list of completed steps for this tab.
+   * @param tabIndex
+   */
+  const getCompletedTabSteps = useCallback(
+    (tabIndex: number) => {
+      return _.uniq(getIn(values, `tabs.${tabIndex}.completedSteps`) || []);
+    },
+    [values],
+  );
+
+  /**
+   * Whenever the tab changes, update the list of steps that have been completed and that are navigable.
+   */
+  useDeepCompareEffect(() => {
+    if (tabs.length > 1) {
+      const completedSteps = getCompletedTabSteps(activeTab);
+      steps.forEach((step, index) => {
+        step.completed = completedSteps.includes(index);
+        step.canGoToStep = completedSteps.includes(index - 1);
+      });
+      if (steps.length) {
+        steps[0].canGoToStep = true;
+      }
+    }
+  }, [activeTab, tabs.length, getCompletedTabSteps, steps]);
+
+  /**
+   * Set the step at the given stepIndex to be completed.
+   */
+  const completeStep = (index: number) => {
+    if (index >= 0 && index < steps.length) {
+      const currentStep = steps[index];
+      currentStep.completed = true;
+      setFieldValue(
+        `tabs.${values.activeTab}.completedSteps`,
+        _.uniq([...getCompletedTabSteps(values.activeTab), index]),
+      );
+      if (index + 1 < steps.length) {
+        steps[index + 1].canGoToStep = true;
+      }
+    }
+  };
+
+  /**
+   * Validate the current step, setting errors if present or completing the step if the step is valid.
+   * @param overrideNameSpace nameSpace to use instead of the original validation nameSpace provided for this step.
+   */
+  const validateCurrentStep = (overrideNameSpace?: string) => {
+    const index = getIn(values, `tabs.${values.activeTab}.activeStep`);
```

Comment:
Just to make sure I am on the same page. `tabs` is referring to the tabs of associated parcels correct? I.E) Parcel 1, 2, 3, 4

Reply:
yeah, in the case of the building form everything will be on the "hidden" tab 0

Reply:
Ahh ok gotchya, makes sense now